core/src/main/scala/org/apache/spark/deploy/mesos/ui/MesosClusterPage.scala

```diff
@@ -68,6 +75,25 @@ private[mesos] class MesosClusterPage(parent: MesosClusterUI) extends WebUIPage(
 
   private def driverRow(state: MesosClusterSubmissionState): Seq[Node] = {
     val id = state.driverDescription.submissionId
+    val masterInfo = parent.scheduler.getSchedulerMasterInfo()
+    val schedulerFwId = parent.scheduler.getSchedulerFrameworkId()
+    val sandboxCol = if (masterInfo.isDefined && schedulerFwId.isDefined) {
+
+      val masterUri = masterInfo.map{info => s"http://${getIp4(info.getIp)}:${info.getPort}"}.get
+      val directory = getTaskDirectory(masterUri, id, state.slaveId.getValue)
```

Comment:
You're making O(N) HTTP requests.  I think that may be an unacceptable performance cost.